### PR TITLE
Fix PG upgrade scripts for 9.4, 9.5

### PR DESCRIPTION
**Pull request by thanodnl**
DESCRIPTION: Fixes an issue that could cause citus_finish_pg_upgrade to fail

Backport of #5076

Important test to pass: `test-11-12_check-pg-upgrade`

### CODE REVIEW

File: src/backend/distributed/sql/udfs/citus_prepare_pg_upgrade/9.5-2.sql [hunk 1]
```diff
@@ -0,0 +1,59 @@
+CREATE OR REPLACE FUNCTION pg_catalog.citus_prepare_pg_upgrade()
+    RETURNS void
+    LANGUAGE plpgsql
+    SET search_path = pg_catalog
+    AS $cppu$
+BEGIN
+    --
+    -- Drop existing backup tables
+    --
+    DROP TABLE IF EXISTS public.pg_dist_partition;
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/sql/udfs/citus_prepare_pg_upgrade/9.5-2.sql:10]: Why do we have these DROPs here but not in 9.4-2?

Comment (thanodnl) [reply src/backend/distributed/sql/udfs/citus_prepare_pg_upgrade/9.5-2.sql:10]: oh, wow! good catch!

Comment (thanodnl) [reply src/backend/distributed/sql/udfs/citus_prepare_pg_upgrade/9.5-2.sql:10]: I guess the drops were introduced in 9.5: https://github.com/citusdata/citus/commit/2a6c867cb023d183fe9a27700081a71ce7e99bde

Comment (SaitTalhaNisanci) [reply src/backend/distributed/sql/udfs/citus_prepare_pg_upgrade/9.5-2.sql:10]: yeah seems like using DROPs would be better on 9.4-2 as well for idempotency, but that seems a bit irrelevant for this PR

**Review by SaitTalhaNisanci — APPROVED** [review]

**Pull request merged**